Dockerfile

```diff
@@ -0,0 +1,9 @@
+FROM golang
+
+COPY . /go/src/github.com/rclone/rclone/
+WORKDIR /go/src/github.com/rclone/rclone/
+
+RUN go build -v
```

Comment:
Ideally this would use `make` rather than `go build` - then the git build tag etc will appear when you do `rclone version`.

There might be a more docker way of doing this as it is easy to express in just go commands.

It would be nice to run some tests here too `make quicktest` takes a couple of minutes and tests all the important things. If that fails then the build should fail.

Reply:
> There might be a more docker way of doing this as it is easy to express in just go commands.

For instance if the build version came in as an environment variable it would be easy to use.

Reply:
I was following the advice you gave me at #3423. Also, I wasn't aware there was a Makefile.
There is no single 'Docker Way' to build any software. I've dockerized software several times using make-based builds, and it's OK to do so, as long as it works.
I would even add that, the more similar all builds are done, the better, since you just want to make the build as reproduceable as possible and the Docker binary to be the same as with other packaging methods. So, if the build tool is `make`, we can switch to it. Whatever makes more sense to you.

About the tests, I'm in for testing builds. Sometime I test at least the most critical parts of the build, to make the build fail loudly. The caveats I see are:
* We have to ensure we cleanup after the tests, to keep image size.
* If it will not add too much time to the build, which is especially relevant for development workflow. The `make` tool is already included in the Golang image.
EDIT: The above caveats do not not apply with multi-stage build, but listing here for completeness.

Also, I've checked and the tests run, but fail. I've tried to run them in the v.1.48.0 tag, and they also fail. If there is something to fix before, or adding them is more involved, it can be left as a future iteration, too. This way, we can tackle #3460 and get familiar with the Automated Builds workflow too ;)

Reply:
How did the tests fail? Can you paste the failure?  If I had to guess I'd say it would be the mount tests failing to make a FUSE mount in the container.

Reply:
Tried reproducing the failure, but now they run fine every time.
They run fine both before and after build. Which is preferred?

Reply:
I normally do the tests before the build.

Are you ready to send the file?

Reply:
Added tests, they pass all the time.
Also, had to statically link the binary, do it won't fail on the Alpine container.